akka-docs/rst/java/stream/stream-rate.rst

```diff
@@ -141,29 +57,124 @@ Rate transformation
+elements that have arrived  while the stats have been calculated.
 
 .. includecode:: ../code/docs/stream/RateTransformationDocTest.java#conflate-summarize
 
-This example demonstrates that such flow's rate is decoupled. The element rate at the start of the flow can be much higher that the element rate at the end of the flow.
+This example demonstrates that such flow's rate is decoupled. The element rate at the start of the flow
+can be much higher that the element rate at the end of the flow.
 
-Another possible use of ``conflate`` is to not consider all elements for summary when producer starts getting too fast. Example below demonstrates how ``conflate`` can be used to implement random drop of elements when consumer is not able to keep up with the producer.
+Another possible use of ``conflate`` is to not consider all elements for summary when producer starts getting too fast.
+Example below demonstrates how ``conflate`` can be used to implement random drop of elements when consumer is not able
+to keep up with the producer.
 
 .. includecode:: ../code/docs/stream/RateTransformationDocTest.java#conflate-sample
 
 Understanding expand
 --------------------
 
-Expand helps to deal with slow producers which are unable to keep up with the demand coming from consumers. Expand allows to extrapolate a value to be sent as an element to a consumer.
+Expand helps to deal with slow producers which are unable to keep up with the demand coming from consumers. Expand
+allows to extrapolate a value to be sent as an element to a consumer.
 
-As a simple use of ``expand`` here is a flow that sends the same element to consumer when producer does not send any new elements.
+As a simple use of ``expand`` here is a flow that sends the same element to consumer when producer does not send any
+new elements.
 
 .. includecode:: ../code/docs/stream/RateTransformationDocTest.java#expand-last
 
-Expand also allows to keep some state between demand requests from the downstream. Leveraging this, here is a flow that tracks and reports a drift between fast consumer and slow producer.
+Expand also allows to keep some state between demand requests from the downstream. Leveraging this, here is a flow
+that tracks and reports a drift between fast consumer and slow producer.
 
 .. includecode:: ../code/docs/stream/RateTransformationDocTest.java#expand-drift
 
-Note that all of the elements coming from upstream will go through ``expand`` at least once. This means that the output of this flow is going to report a drift of zero if producer is fast enough, or a larger drift otherwise.
+Note that all of the elements coming from upstream will go through ``expand`` at least once. This means that the
+output of this flow is going to report a drift of zero if producer is fast enough, or a larger drift otherwise.
+
+
+.. _async-stream-buffers-java:
+
+Buffers for asynchronous stages
```

Comment:
So what's to be decided here? Just if we move it down here or not?
Seems to me keeping it at the top of this page would make understanding conflate simpler, as the protocol is a bit explained here already?